api/src/main/java/org/pmiops/workbench/audit/AbstractAuditableEvent.java

```diff
@@ -0,0 +1,69 @@
+package org.pmiops.workbench.audit;
+
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Payload.JsonPayload;
+import com.google.cloud.logging.Severity;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public abstract class AbstractAuditableEvent {
+  private static final MonitoredResource MONITORED_RESOURCE =
+      MonitoredResource.newBuilder("global").build();
+  private static final String LOG_NAME = "action-audit-test-3";
+
+  public abstract long timestamp();
```

Comment:
I tried that initially, but the Logging library choked on it when building the `JsonPayload`. So it would just be an exercise in converting it and immediately switching back to `long`. 

I'm debating even keeping it, since there are at least two other timestamps associated with the LogEntry that I get for free.